File: algorithms/sort-algorithms/counting-sort/countingSort.js
function countingSort(arr) {
  let largest = 0;
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] < 0) return [];
    if (arr[i] > largest) largest = arr[i];
  }
  const tracker = [...Array(largest + 1)].fill(0);
  const final = [...Array(arr.length - 1)].fill(0);
  arr.forEach(num => tracker[num]++);
  tracker.forEach((num, i) => {
    if (i) tracker[i] += tracker[i - 1];
  });
  arr.forEach((num, i) => {
    if (!!tracker[num]) {
      final[tracker[num] - 1] = num;
      tracker[num] -= 1;
    }
  });
  return final;
}

File: algorithms/sort-algorithms/counting-sort/counting_sort.py
def counting_sort(l):
    largest = 0
    for num in l:
        if num < 0:
            return []
        elif num > largest:
            largest = num
    tracker = [0 for i in range(largest + 1)]
    final = [0 for i in range(len(l))]
    for num in l:
        tracker[num] += 1
    for i in range(len(tracker)):
        if i > 0:
            tracker[i] += tracker[i - 1]
    for num in l:
        if tracker[num] > 0:
            final[tracker[num] - 1] = num
            tracker[num] -= 1
    return final
